cdap-security/src/main/java/co/cask/cdap/security/impersonation/DefaultImpersonator.java

```diff
@@ -63,8 +63,10 @@ public DefaultImpersonator(CConfiguration cConf, UGIProvider ugiProvider) {
   public <T> T doAs(NamespacedEntityId entityId, Callable<T> callable,
                     ImpersonatedOpType impersonatedOpType) throws Exception {
     UserGroupInformation ugi = getUGI(entityId, impersonatedOpType);
-    LOG.debug("Performing doAs with UGI {} for entity {} and impersonation operation type", ugi, entityId,
-              impersonatedOpType);
+    if (kerberosEnabled && !NamespaceId.SYSTEM.equals(entityId.getNamespaceId())) {
```

Comment:
I think its cleaner to check `if (!UserGroupInformation.getCurrentUser().equals(ugi)) {`.